lib/ansible/modules/user.py

```diff
@@ -634,11 +634,11 @@ def create_user_useradd(self):
 
         if self.home is not None:
             # If the specified path to the user home contains parent directories that
-            # do not exist, first create the home directory since useradd cannot
+            # does not exist, first create the home directory since useradd cannot
             # create parent directories
             parent = os.path.dirname(self.home)
             if not os.path.isdir(parent):
-                self.create_homedir(self.home)
+                os.makedirs(parent)
```

Comment:
Creating dirs recursively has a number of security problems and we generally avoid this.

Reply:
ok, there is nothing about that in python doc, Thanks